netket/utils/symmetry/group.py

```diff
@@ -0,0 +1,265 @@
+        Calculates the character table using Burnside's algorithm.
+
+        Each row of the output lists the characters of one irrep in the order the
+        conjugacy classes are listed in `self.conjugacy_classes`.
+
+        Assumes that `Identity() == self[0]`, if not, the sign of some characters
+        may be flipped. The irreps are sorted by dimension.
+        """
+        classes, _, _ = self.conjugacy_classes
+        class_sizes = classes.sum(axis=1)
+        # Construct a random linear combination of the class matrices c_S
+        #    (c_S)_{RT} = #{r,s: r \in R, s \in S: rs = t}
+        # for conjugacy classes R,S,T, and a fixed t \in T.
+        #
+        # From our oblique times table it is easier to calculate
+        #    (d_S)_{RT} = #{r,t: r \in R, t \in T: rs = t}
+        # for a fixed s \in S. This is just `product_table == s`, aggregrated
+        # over conjugacy classes. c_S and d_S are related by
+        #    c_{RST} = |S| d_{RST} / |T|;
+        # since we only want a random linear combination, we forget about the
+        # constant |S| and only divide each column through with the appropriate |T|
+        class_matrix = (
+            classes @ np.random.uniform(size=len(self))[self.product_table] @ classes.T
+        )
+        class_matrix /= class_sizes
+
+        # The vectors |R|\chi(r) are (column) eigenvectors of all class matrices
+        # the random linear combination ensures (with probability 1) that
+        # none of them are degenerate
+        _, table = np.linalg.eig(class_matrix)
+        table = table.T / class_sizes
+
+        # Normalise the eigenvectors by orthogonality: \sum_g |\chi(g)|^2 = |G|
+        norm = np.sum(np.abs(table) ** 2 * class_sizes, axis=1, keepdims=True) ** 0.5
+        table /= norm
+        table /= _cplx_sign(table[:, 0])[:, np.newaxis]  # ensure correct sign
+        table *= len(self) ** 0.5
+
+        # Sort lexicographically, ascending by first column, descending by others
+        sorting_table = np.column_stack((table.real, table.imag))
+        sorting_table[:, 1:] *= -1
+        sorting_table = comparable(sorting_table)
+        _, indices = np.unique(sorting_table, axis=0, return_index=True)
+        table = table[indices]
+
+        # Get rid of annoying nearly-zero entries
+        table = prune_zeros(table)
+
+        return table
+
+    def character_table(self) -> Array:
+        """
+        Calculates the character table using Burnside's algorithm.
+
+        Each row of the output lists the characters of all group elements for one irrep,
+        i.e. self.character_table()[i,g] gives :math:`\chi_i(g)`.
+
+        Assumes that `Identity() == self[0]`, if not, the sign of some characters
+        may be flipped. The irreps are sorted by dimension.
+        """
+        _, _, inverse = self.conjugacy_classes
+        CT = self.character_table_by_class
+        return CT[:, inverse]
+
+    def character_table_readable(self) -> Tuple[List[str], Array]:
```

Comment:
maybe we should pick a more explicative name...

Reply:
It's kinda hard, because the whole point is that it produces something that is human-readable on the console to help the user find the irrep they need. It's not useful in production scripts

Reply:
Maybe make a data class `CharacterTable` with custom `__repr__`?

Reply:
Could do but it would have a very limited use case, and would duplicate a lot of data from Group (everything about conjugacy classes, quite a bit about the group elements, etc.), so not a fan